templates/CRM/MembershipExtras/Page/NextPeriodTab.tpl

```diff
@@ -1,22 +1,83 @@
 <script>
 var recurringContributionID = {$recurringContributionID};
+var financialTypes = JSON.parse('{$financialTypes|@json_encode}');
 
 {literal}
   CRM.$(function () {
+
     CRM.$('.remove-next-period-line-button').each(function () {
       CRM.$(this).click(function (e) {
         e.preventDefault();
-
         var itemData = CRM.$(this).closest('tr').data('item-data');
         showNextPeriodLineItemRemovalConfirmation(itemData);
 
-        CRM.$('#periodsContainer').on('crmLoad', function(event, data) {
-          CRM.$('#tab_next a').click();
-        });
+        window.CompucorpMembershipExtras_selectedTab = 'next';
```

Comment:
I don't think this should happen here only if the remove icon is clicked. I think we should add this logic to actually clicking the tabs instead and move it to EditContributionRecurLineItems.tpl.

Eg.
```javascript
var selectedTab = window.CompucorpMembershipExtras_selectedTab || 'current';

CRM.$(function($) {
  CRM.$('#tab_current').click(function () {
    window.CompucorpMembershipExtras_selectedTab = 'current';
  });

  CRM.$('#tab_next').click(function () {
    window.CompucorpMembershipExtras_selectedTab = 'next';
  });

  var tabIndex = $('#tab_' + selectedTab).prevAll().length;
  $("#periodsContainer").tabs({active: tabIndex});
  $(".crm-tab-button").addClass("ui-corner-bottom");
});
```

Reply:
The problem with this approach is that whatever tab the user closed the modal popup on will be reopened if the user reopens the popup without refreshing the page. Is that a desired effect?

Reply:
Hmm, good point. Maybe if I open the same recurring contribution it could make sense, but it might be awkward if I open another recurring contribution. What if we add an attribute to the dialog's markup to store the value instead? Something like:

```javascript
var selectedTab = CRM.$(#periodsContainer).closest('.ui-dialog-content').data('selectedTab') || 'current';
...
CRM.$(#periodsContainer).closest('.ui-dialog-content').data('selectedTab', 'current');
...
CRM.$(#periodsContainer).closest('.ui-dialog-content').data('selectedTab', 'next');
```

That way, every time the dialog is opened, the value would be empty.